kyo-prelude/shared/src/test/scala/kyo/AspectTest.scala

```diff
@@ -267,4 +201,115 @@ class AspectTest extends Test:
         }
     }
 
+    "non-Const type parameters" - {
+        case class Wrapped[+A, B](value: A, meta: B) derives CanEqual
+        case class Container[+A](value: A, meta: String) derives CanEqual
+
+        // TODO SIGSEGV in Scala Native
```

Comment:
It seems a bug in Scala Native since JVM and JS pass. I couldn't isolate it but all these tests fail in Native.

Reply:
I've changed the effect to make `Cut` the generic function directly and now all tests are failing in Native. It seems the issue is the use of the generic functions. I'll open a ticket to follow up.